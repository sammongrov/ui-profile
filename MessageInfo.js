import React, { Component } from 'react';
import { TouchableOpacity, ScrollView } from 'react-native';
import moment from 'moment';
import { Colors } from '@ui/theme_default';

import { styles } from 'react-native-theme';
import { Actions } from 'react-native-router-flux';
import PropTypes from 'prop-types';
import { Screen, NavBar, Icon, ProfileList } from '@ui/components';
// import { Colors } from '@ui/theme_default';
// import DBManager from '../app/DBManager';
import {DBManager} from 'app-module';
// const styles = StyleSheet.create({});
import {Application} from '@mongrov/config';

export default class MessageInfo extends Component {
  constructor(props) {
    super(props);
    const { messageId } = this.props;
    const { chat } = DBManager._taskManager;
    chat.getMessageInfo(messageId, this.messageInfo);
    console.log('messageInfo MESSAGE ID', messageId);
    this.state = {
      memberList: [],
    };
  }

  messageInfo = (error, messageInfo) => {
    if (!error) {
      const memberList = [];

      for (let i = 0; i < messageInfo.length; i += 1) {
        const _messageInfo = messageInfo[i];
        const users = {
          id: _messageInfo.user._id,
          title: _messageInfo.user.name,
          subTitle: `@${_messageInfo.user.username}`,
          avatar_url: `${Application.urls.SERVER_URL}/avatar/${_messageInfo.user.username}`,
          date: moment(_messageInfo.ts).format('MMM Do YY'),
          time: moment(_messageInfo.ts).format('LT'),
        };
        memberList.push(users);
      }
      this.setState({
        memberList,
      });
    }
  };

  render() {
    const { memberList } = this.state;
    return (
      <Screen>
        <NavBar
          leftComponent={
            <TouchableOpacity
              onPress={Actions.pop}
              style={[styles.navSideButtonDimension, styles.alignJustifyCenter]}
            >
              <Icon
                name="chevron-left"
                type="material-community"
                color={Colors.NAV_ICON}
                size={36}
              />
            </TouchableOpacity>
          }
          titleText="Read By"
        />
        <ScrollView showsVerticalScrollIndicator={false}>
          <ProfileList list={memberList} />
        </ScrollView>
      </Screen>
    );
  }
}

MessageInfo.propTypes = {
  messageId: PropTypes.string,
};

MessageInfo.defaultProps = {
  messageId: '',
};
