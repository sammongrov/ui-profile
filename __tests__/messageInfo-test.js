import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DbManager from '../../app/DBManager';
import MessageInfo from '../MessageInfo';

configure({ adapter: new Adapter() });

jest.mock('../../app/DBManager', () => {
  const dbManager = {
    _taskManager: {},
  };
  return dbManager;
});

const messageInfo = [
  {
    user: { _id: 'id123456', name: 'user1', username: '1user' },
    ts: 1547884787152,
  },
  {
    user: { _id: 'id123457', name: 'user2', username: '2user' },
    ts: 1547884787052,
  },
  {
    user: { _id: 'id123455', name: 'user3', username: '3user' },
    ts: 1547884786053,
  },
];

DbManager._taskManager.chat = {
  getMessageInfo: jest.fn((id, cb) => cb(null, messageInfo)),
};

it('MessageInfo renders correctly with messageId', () => {
  const tree = renderer.create(<MessageInfo messageId="m985OP253fx" />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('messageInfo method is called', () => {
  it('with non empty array', () => {
    const tree = shallow(<MessageInfo messageId="m985OP253fx" />);
    const treeInstance = tree.instance();
    treeInstance.messageInfo(null, messageInfo);
    expect(tree.state().memberList.length).toBe(3);
    expect(tree.state().memberList[0].id).toMatch(messageInfo[0].user._id);
    expect(tree.state().memberList[1].title).toMatch(messageInfo[1].user.name);
    expect(tree.state().memberList[2].subTitle).toMatch(`@${messageInfo[2].user.username}`);
  });

  it('with an empty array', () => {
    const tree = shallow(<MessageInfo messageId="m985OP253fx" />);
    const treeInstance = tree.instance();
    treeInstance.messageInfo(null, []);
    expect(tree.state().memberList.length).toBe(0);
  });

  it('with an error', () => {
    const error = new Error('test error');
    const tree = shallow(<MessageInfo messageId="m985OP253fx" />);
    const treeInstance = tree.instance();
    treeInstance.messageInfo(error, []);
    expect(tree.state().memberList.length).toBe(0);
  });
});
