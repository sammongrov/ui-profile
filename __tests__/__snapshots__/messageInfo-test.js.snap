// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInfo renders correctly with messageId 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2E88FF",
                    "fontSize": 36,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="material-community"
            >
              ÔÖÅ
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            Read By
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <RCTScrollView
        showsVerticalScrollIndicator={false}
      >
        <View>
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
